src/share/classes/com/sun/javatest/regtest/agent/MainWrapper.java

```diff
@@ -70,6 +72,18 @@ public static void main(String[] args) {
 
         // RUN JAVA IN ANOTHER THREADGROUP
         MainThreadGroup tg = new MainThreadGroup();
+
+        Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
+            synchronized (MainWrapper.class) {
+                if (e instanceof ThreadDeath)
+                    return;
+                e.printStackTrace(System.err);
+                if (globalUncaughtThrowable == null) {
+                    globalUncaughtThrowable = e;
+                }
+                AStatus.failed(MAIN_THREW_EXCEPT + e).exit();
```

Comment:
I think we shouldn't be failing and exiting here because the uncaught exception might have been thrown when the test is still in progress. Just keeping track of the uncaught exception(s) I think should be enough, so that that state can then be used after the test completes, to decide whether to fail the test. This is then similar to what we do in the case where no test thread factory is set.

Reply:
I copied the handler from MainThreadGroup::uncaughtException. I want to keep them consistent. (Eventually, we are going to get rid of MainGroup.) Probably, it is needed to file a bug not to fail in the MainThreadGroup::uncaughtException and check exceptions after the test passes.

Reply:
Hello Leonid,

> I copied the handler from MainThreadGroup::uncaughtException. I want to keep them consistent.

I hadn't noticed MainThreadGroup does it this way. Having it consistent then sounds good to me.